### PR TITLE
Promote DNS e2e verifying PQDN resolution for services

### LINKED ISSUE #73425: Tracking Issue - Conformance Coverage for Networking
**What would you like to be added**:
This is a place holder to track Conformance Coverage on the Topic.
This issue should remain open until all related work is accomplished in the k/k repo
This issue will contain analysis of the coverage. Existing tests and additional tests requested with links to those issues/PR's

**Why is this needed**:
Specifically address CNI networking, identify tests that tests pod-pod communication

/area conformance

### PR DESCRIPTION
**Pull request by mgdevstack**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug

/kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Promotes DNS e2e which is verifying PQDN resolution behavior for services.

**Which issue(s) this PR fixes**:

Fixes #73425

**Special notes for your reviewer**:
- Maybe failing for windows.
- Not Flaking
- Promoting to Conformance as it verifies important functional scenario of resolving PQDN for services.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```
/area conformance
/sig testing
@kubernetes/sig-network-pr-reviews

### CODE REVIEW

**Comment by brahmaroutu** [comment]
@smarterclayton @nicksardo    Can you please review and lgtm to promote this PR.

**Comment by bclau** [comment]
-1
These tests will not pass on Windows, which is why they were separated from the original test in the first place https://github.com/kubernetes/kubernetes/pull/70156. If you really want to promote this to Conformance, at least add the [LinuxOnly] tag to the test name.

File: test/e2e/network/dns.go [hunk 1]
```diff
@@ -162,7 +162,12 @@ var _ = SIGDescribe("DNS", func() {
 		validateDNSResults(f, pod, append(wheezyFileNames, jessieFileNames...))
 	})
 
-	It("should resolve DNS of partial qualified names for services ", func() {
+	/*
+		Release: v1.14
```
Comment (spiffxp) [inline test/e2e/network/dns.go]:
```suggestion
		Release: v1.15
```

**Comment by spiffxp** [comment]
/lgtm cancel
for same reasons as https://github.com/kubernetes/kubernetes/pull/74977

File: test/e2e/network/dns.go [hunk 2]
```diff
@@ -162,7 +162,12 @@ var _ = SIGDescribe("DNS", func() {
...
+		Testname: DNS, PQDN for services
+		Description: Create a headless service and normal service. Both the services MUST be able to resolve partial qualified DNS entries of their service endpoints by serving A records and SRV records e.g. serviceName, serviceName.namespace, _http._tcp.serviceName.namespace.svc. at that endpoints pointing to reguler service IP.
+	*/
+	framework.ConformanceIt("should resolve DNS of partial qualified names for services ", func() {
```
Comment (johnbelamaric) [inline test/e2e/network/dns.go]: Tag as LinuxOnly and include the reason why Windows cannot resolve PQDN (link to KEP preferably).

Comment (mgdevstack) [reply test/e2e/network/dns.go]: Sure, Creating new PR for LinuxOnly tagging.
/cc @bclau for KEP

File: test/e2e/network/dns.go (hunk 1 above)
Comment (mgdevstack) [reply test/e2e/network/dns.go]: Done.

**Comment by mgdevstack** [comment]
/hold 
as per https://github.com/kubernetes/kubernetes/pull/74980#issuecomment-475830486

**Comment by mgdevstack** [comment]
Needs rebase once #75758 merged.

**Comment by mgdevstack** [comment]
Closing this PR as it's Promotion can be included in another related PR https://github.com/kubernetes/kubernetes/pull/74977 along with another e2e `should support configurable pod DNS nameservers`

Note: _Seperate PRs(#74980, #74977, #74985) were created to promote these dns specific E2Es because of their non-visible clarity wrt. Windows platform. With series of discussions held in these PRs and merging https://github.com/kubernetes/kubernetes/pull/75758, it's good to club them into single PR i.e. https://github.com/kubernetes/kubernetes/pull/74977_

**Comment by mgdevstack** [comment]
Another valid reason of closing it that, it's good to complete all `// TODO:` before promoting them.

**Pull request closed without merging**